src/main/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDbBase.java

```diff
@@ -364,9 +364,16 @@ public ListTablesResult listTables(String exclusiveStartTableName, Integer limit
                 innerExclusiveStartTableName = rawResults.getLastEvaluatedTableName();
             } while (!(tableNames.size() >= limit || innerExclusiveStartTableName == null));
 
-            return new ListTablesResult()
-                .withLastEvaluatedTableName(tableNames.isEmpty() ? null : tableNames.get(tableNames.size() - 1))
-                .withTableNames(tableNames.size() > limit ? tableNames.subList(0, limit) : tableNames);
+            // if there's more results left to go through, populate the lastEvaluatedTableName so the client can
+            // page through for more results.
+            if (tableNames.size() > limit || innerExclusiveStartTableName != null) {
+                return new ListTablesResult()
+                    .withLastEvaluatedTableName(tableNames.get(Math.min(tableNames.size() - 1, limit - 1)))
+                    .withTableNames(tableNames.size() > limit ? tableNames.subList(0, limit) : tableNames);
```

Comment:
I feel like introducing a local variable could make this code a little simpler:
```
final List<String> tableNames = tableNames.size() > limit ? tableNames.subList(0, limit) : tableNames;
return new ListTablesResult().withLastEvaluatedTableName(Iterables.getLast(tableNames)).withTableNames(tableNames)
```

Reply:
that snippet doesn't hold the contract of only setting the last evaluated name if the result set is not full, unfortunately, as you pointed out in a previous review. If we always set lastEvaluatedTableName on the last non-full page of results, we're requiring the clients to pull one more page of empty results from the actual last table name.

Reply:
oh you mean within this if block when we know we have to set the last table name. Makes sense, will update

Reply:
that said, `Iterables.getLast(tableNames)` is insufficient if `tableNames.size() > limit`, in which case we'd drop the items between `limit` and `tableNames.size()` to the ground. Will keep that snippet as is (`.withLastEvaluatedTableName(tableNames.get(Math.min(tableNames.size() - 1, limit - 1)))`)